applications/registration-react/src/pages/dataset-registration-page/form-sample/form-sample.component.jsx

```diff
@@ -26,7 +26,7 @@ export const renderSamples = componentProps => {
                   fields.remove(index);
                   asyncValidate(
                     fields.getAll(),
-                    null,
+                    componentProps.dispatch,
```

Comment:
reduxform har  en hook asyncValidate, som du kan bruke. Men det mener ikke, at du må bruke samme funksjon i din egen onclick handler.

Så, hvis du deler din nåværende asyncValidate i to - asyncValidate(values, dispatch) og patchDataset(datasetId, values, dispatch), da kan du kalle den ny patchDataset rett fra onClick, uten å bruke den redux-forms spesifikk asyncValidate format.

Det hva skjer i asyncValidate funksjon er rett og slett kriminell. Begynne med faktorering. Målet av den hele modul er til å håndtere dataset form patch. Bevege under riktig plass under dataset - lib er generell, gjelder til alle moduler.